lib/toopher_api.rb

```diff
@@ -44,13 +51,109 @@ class UnknownTerminalError < ToopherApiError
 class PairingDeactivatedError< ToopherApiError
 end
 
+class SignatureValidationError< ToopherApiError
+end
+
+class ToopherIframe
+  DEFAULT_IFRAME_TTL = 300
+  IFRAME_VERSION = '2'
+
+  def initialize(key, secret, options, base_url = DEFAULT_BASE_URL)
+    key.empty? and raise ArgumentError, "Toopher consumer key cannot be empty!"
+    secret.empty? and raise ArgumentError, "Toopher consumer secret cannot be empty!"
+
+    @oauth_options = options.merge(:site => base_url, :scheme => :query_string)
+    @oauth_consumer = OAuth::Consumer.new(key, secret, @oauth_options)
+    @base_url = base_url
+  end
+
+  def get_user_management_url(username, reset_email, **kwargs)
+    ttl = kwargs.delete(:ttl) || DEFAULT_IFRAME_TTL
+    params = {
+      :username => username,
+      :reset_email => reset_email,
+      :v => IFRAME_VERSION,
+    }
+    params.merge!(kwargs)
+    get_oauth_signed_url('web/manage_user', ttl, params)
+  end
+
+  def get_authentication_url(username, reset_email, request_token, action_name='Log In', requester_metadata='None', **kwargs)
+    kwargs[:allow_inline_pairing] = true if kwargs[:allow_inline_pairing].nil?
+    kwargs[:automation_allowed] = true if kwargs[:automation_allowed].nil?
+    kwargs[:challenge_required] ||= false
```

Comment:
The default values for `allow_inline_pairing`, `automation_allowed` and `challenge_required` are set in the AuthenticationRequestInitiationHandler so you don't need to worry about setting these values here. (I'll make the corresponding changes in toopher-python.)